test/performance/io/format_fastq_benchmark.cpp

```diff
@@ -0,0 +1,230 @@
+// -----------------------------------------------------------------------------------------------------
+// Copyright (c) 2006-2019, Knut Reinert & Freie Universität Berlin
+// Copyright (c) 2016-2019, Knut Reinert & MPI für molekulare Genetik
+// This file may be used, modified and/or redistributed under the terms of the 3-clause BSD-License
+// shipped with this file and also available at: https://github.com/seqan/seqan3/blob/master/LICENSE.md
+// -----------------------------------------------------------------------------------------------------
+
+//!\author: Clemens Cords <clemens.cords@fu-berlin.de>
+
+#include <benchmark/benchmark.h>
+#include <cstring>
+#include <fstream>
+
+#include <seqan3/test/seqan2.hpp>
+#if SEQAN3_HAS_SEQAN2
+#include <seqan/seq_io.h>
+#endif
+
+#include <seqan3/alphabet/nucleotide/dna5.hpp>
+#include <seqan3/alphabet/quality/phred42.hpp>
+#include <seqan3/io/sequence_file/input.hpp>
+#include <seqan3/io/sequence_file/output.hpp>
+#include <seqan3/range/views/to_char.hpp>
+#include <seqan3/std/filesystem>
+#include <seqan3/test/performance/sequence_generator.hpp>
+#include <seqan3/test/tmp_filename.hpp>
+
+using namespace seqan3;
+
+constexpr unsigned default_seed = 1234u;
+
+// ============================================================================
+// generate fastq file
+// ============================================================================
+constexpr size_t default_sequence_length = 50; //length of nucleotide and quality sequence
+
+std::string generate_fastq_file(size_t const entries_size)
+{
+    std::stringstream file{};
+    std::string const id{"@name"};
+
+    auto seed = default_seed;
+
+    for (size_t i = 0; i < entries_size; ++i, ++seed)
+    {
+        auto seq = test::generate_sequence<dna5>(default_sequence_length, 0, seed);
+        auto seq_as_chars = seq | views::to_char;
+
+        auto qual = test::generate_sequence<phred42>(default_sequence_length, 0, seed);
+        auto qual_as_chars = qual | views::to_char;
+
+        file << id << '\n' << std::string(seq_as_chars.begin(), seq_as_chars.end()) << '\n' << '+' << '\n' << std::string(qual_as_chars.begin(), qual_as_chars.end()) << '\n';
```

Comment:
views::to_char threw that implicit conversion from views to containers is now deprecated, I couldn't think of a better way to append and convert the ranges to std::string